sdk/core/core/inc/az_http_pipeline.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_HTTP_PIPELINE_H
+#define AZ_HTTP_PIPELINE_H
+
+#include <az_contract.h>
+#include <az_http_request.h>
+#include <az_iter_data.h>
+#include <az_pair.h>
+#include <az_span.h>
+#include <az_str.h>
+
+#include <az_http_policy.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef struct {
+  az_http_policy policies[10];
+  int num_policies;
+  int pipeline_stage;
+
+  //Each policy calls next
+  typedef az_result (*az_http_pipeline_next)(az_http_pipeline const * pipeline, az_http_request const * p_request);
```

Comment:
typedef inside struct?

Reply:
I think C99 doesn't allow this.

Reply:
I didn't get errors from it.  However I've changed from doing this to using a nextpolicy method.  Cleaner given the constraints C99 makes on us